collector/mdadm_linux.go

```diff
@@ -18,199 +18,141 @@ package collector
 import (
 	"fmt"
 	"io/ioutil"
+	"math"
 	"os"
-	"regexp"
 	"strconv"
 	"strings"
 
+	"syscall"
+	"unsafe"
+
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/log"
 )
 
-var (
-	statuslineRE             = regexp.MustCompile(`(\d+) blocks .*\[(\d+)/(\d+)\] \[[U_]+\]`)
-	raid0lineRE              = regexp.MustCompile(`(\d+) blocks .*\d+k chunks`)
-	buildlineRE              = regexp.MustCompile(`\((\d+)/\d+\)`)
-	unknownPersonalityLineRE = regexp.MustCompile(`(\d+) blocks (.*)`)
-	raidPersonalityRE        = regexp.MustCompile(`raid[0-9]+`)
+type (
+	mdStatus struct {
+		name string
+		//active       bool
+		active       float64
+		disksActive  float64
+		disksFailed  float64
+		disksMissing float64
+		disksSpare   float64
+		bytesTotal   float64
+		bytesSynced  float64
+	}
+
+	mdu_array_info struct {
```

Comment:
Go naming convention is to use camel case instead of underscores.

Also, these structures should be automatically generated in `x/sys/unix` for maximum portability.